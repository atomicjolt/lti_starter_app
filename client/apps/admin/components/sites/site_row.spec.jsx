import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer, { act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import SiteRow from './site_row';

const mockStore = configureStore([]);
const store = mockStore({
  settings: {},
});

describe('sites list row', () => {
  let result;
  let instance;

  const site = { url: 'http://www.example.com' };
  const deleteSite = () => {};

  // https://medium.com/@amanverma.dev/mocking-create-portal-to-utilize-react-test-renderer-in-writing-snapshot-uts-c49773c88acd
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element) => element);
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });

  beforeEach(() => {
    result = TestRenderer.create(
      <Provider store={store}>
        <SiteRow
          site={site}
          deleteSite={deleteSite}
        />
      </Provider>
    );
    instance = result.root;
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the first button onclick event', () => {
    expect(result).toMatchSnapshot();
    const buttons = instance.findAllByType('button');
    act(() => {
      buttons[0].props.onClick();
    });
    expect(result).toMatchSnapshot();
  });

  it('handles the second button onclick event', () => {
    let buttons = instance.findAllByType('button');
    act(() => {
      buttons[1].props.onClick();
    });

    buttons = instance.findAllByType('button');
    const button = buttons.find((b) => b.children[0] === 'Cancel');
    expect(button).toBeTruthy();
  });
});
