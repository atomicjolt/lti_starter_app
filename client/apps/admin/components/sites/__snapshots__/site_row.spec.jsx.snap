// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sites list row matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SiteRow
    site={
      Object {
        "url": "http://www.example.com",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          http://www.example.com
          <span
            className="c-alert c-alert--danger"
            style={
              Object {
                "fontSize": "10px",
              }
            }
          >
            OAuth key and/or secret not configured
          </span>
        </td>,
        <td>
          <button
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "border": "none",
                "color": "grey",
                "cursor": "pointer",
                "fontSize": "1.5em",
              }
            }
          >
            <i
              className="i-settings"
            />
          </button>
          <Connect(SiteModal)
            closeModal={[Function]}
            isOpen={false}
            site={
              Object {
                "url": "http://www.example.com",
              }
            }
          />
        </td>,
        <td>
          <button
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "border": "none",
                "color": "grey",
                "cursor": "pointer",
                "fontSize": "1.5em",
              }
            }
          >
            <i
              className="i-delete"
            />
          </button>
          <Connect(SiteModal)
            closeModal={[Function]}
            isOpen={false}
            site={
              Object {
                "url": "http://www.example.com",
              }
            }
          />
        </td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "http://www.example.com",
            <span
              className="c-alert c-alert--danger"
              style={
                Object {
                  "fontSize": "10px",
                }
              }
            >
              OAuth key and/or secret not configured
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "http://www.example.com",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "OAuth key and/or secret not configured",
              "className": "c-alert c-alert--danger",
              "style": Object {
                "fontSize": "10px",
              },
            },
            "ref": null,
            "rendered": "OAuth key and/or secret not configured",
            "type": "span",
          },
        ],
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "border": "none",
                  "color": "grey",
                  "cursor": "pointer",
                  "fontSize": "1.5em",
                }
              }
            >
              <i
                className="i-settings"
              />
            </button>,
            <Connect(SiteModal)
              closeModal={[Function]}
              isOpen={false}
              site={
                Object {
                  "url": "http://www.example.com",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="i-settings"
              />,
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "transparent",
                "border": "none",
                "color": "grey",
                "cursor": "pointer",
                "fontSize": "1.5em",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "i-settings",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "closeModal": [Function],
              "isOpen": false,
              "site": Object {
                "url": "http://www.example.com",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "border": "none",
                  "color": "grey",
                  "cursor": "pointer",
                  "fontSize": "1.5em",
                }
              }
            >
              <i
                className="i-delete"
              />
            </button>,
            <Connect(SiteModal)
              closeModal={[Function]}
              isOpen={false}
              site={
                Object {
                  "url": "http://www.example.com",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="i-delete"
              />,
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "transparent",
                "border": "none",
                "color": "grey",
                "cursor": "pointer",
                "fontSize": "1.5em",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "i-delete",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "closeModal": [Function],
              "isOpen": false,
              "site": Object {
                "url": "http://www.example.com",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            http://www.example.com
            <span
              className="c-alert c-alert--danger"
              style={
                Object {
                  "fontSize": "10px",
                }
              }
            >
              OAuth key and/or secret not configured
            </span>
          </td>,
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "border": "none",
                  "color": "grey",
                  "cursor": "pointer",
                  "fontSize": "1.5em",
                }
              }
            >
              <i
                className="i-settings"
              />
            </button>
            <Connect(SiteModal)
              closeModal={[Function]}
              isOpen={false}
              site={
                Object {
                  "url": "http://www.example.com",
                }
              }
            />
          </td>,
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "border": "none",
                  "color": "grey",
                  "cursor": "pointer",
                  "fontSize": "1.5em",
                }
              }
            >
              <i
                className="i-delete"
              />
            </button>
            <Connect(SiteModal)
              closeModal={[Function]}
              isOpen={false}
              site={
                Object {
                  "url": "http://www.example.com",
                }
              }
            />
          </td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "http://www.example.com",
              <span
                className="c-alert c-alert--danger"
                style={
                  Object {
                    "fontSize": "10px",
                  }
                }
              >
                OAuth key and/or secret not configured
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "http://www.example.com",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "OAuth key and/or secret not configured",
                "className": "c-alert c-alert--danger",
                "style": Object {
                  "fontSize": "10px",
                },
              },
              "ref": null,
              "rendered": "OAuth key and/or secret not configured",
              "type": "span",
            },
          ],
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "border": "none",
                    "color": "grey",
                    "cursor": "pointer",
                    "fontSize": "1.5em",
                  }
                }
              >
                <i
                  className="i-settings"
                />
              </button>,
              <Connect(SiteModal)
                closeModal={[Function]}
                isOpen={false}
                site={
                  Object {
                    "url": "http://www.example.com",
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="i-settings"
                />,
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "transparent",
                  "border": "none",
                  "color": "grey",
                  "cursor": "pointer",
                  "fontSize": "1.5em",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "i-settings",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "closeModal": [Function],
                "isOpen": false,
                "site": Object {
                  "url": "http://www.example.com",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "border": "none",
                    "color": "grey",
                    "cursor": "pointer",
                    "fontSize": "1.5em",
                  }
                }
              >
                <i
                  className="i-delete"
                />
              </button>,
              <Connect(SiteModal)
                closeModal={[Function]}
                isOpen={false}
                site={
                  Object {
                    "url": "http://www.example.com",
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="i-delete"
                />,
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "transparent",
                  "border": "none",
                  "color": "grey",
                  "cursor": "pointer",
                  "fontSize": "1.5em",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "i-delete",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "closeModal": [Function],
                "isOpen": false,
                "site": Object {
                  "url": "http://www.example.com",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
